pkg/ossm/operator.go

```diff
@@ -0,0 +1,139 @@
+	t.Run("test_ossm_operator_deploy_infra_nodes", func(t *testing.T) {
+		defer util.RecoverPanic(t)
+		util.Log.Info("Testing: Run OSSM Operator on infra nodes")
+		//Get and pick one worker node that does not have already installed the istio operator
+		workername := pickWorkerNode(t)
+		//Label the worker node as infra
+		_, err := util.Shell(`oc label node %s node-role.kubernetes.io/infra=`, workername)
+		if err != nil {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		_, err = util.Shell(`oc label node %s node-role.kubernetes.io=infra`, workername)
+		if err != nil {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		//verify that the worker node is labeled as infra
+		name, err := util.Shell(`oc get nodes -l node-role.kubernetes.io/infra= -o jsonpath='{.items[0].metadata.name}'`)
+		if err != nil {
+			t.Fatalf("Failed to get the infra node name")
+		}
+		if name != workername {
+			t.Fatalf("Failed to label the worker node as infra")
+		}
+		//Taint the node. The only validation to check this is the output message
+		response, err := util.Shell(`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule node-role.kubernetes.io/infra=reserved:NoExecute`)
+		if err != nil {
+			t.Fatalf("Failed to taint the infra node")
+		}
+		if !strings.Contains(response, "tainted") {
+			t.Fatalf("Failed to taint the infra node")
+		}
+		//Edit the subscription to add the infra node to the node selector
+		_, err = util.Shell(`oc patch subscription servicemeshoperator -n openshift-operators --type merge -p '{"spec":{"config":{"nodeSelector":{"node-role.kubernetes.io/infra":""},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","value":"reserved"},{"effect":"NoExecute","key":"node-role.kubernetes.io/infra","value":"reserved"}]}}}'`)
+
+		if err != nil {
+			t.Fatalf("Failed to patch the subscription")
+		}
+		//Verify that the operator pod is running on the infra node
+		podname, err := util.Shell(`oc get pods -n openshift-operators -l name=istio-operator --field-selector spec.nodeName=%s -o jsonpath='{.items[0].metadata.name}'`, workername)
+		if err != nil {
+			time.Sleep(time.Duration(20) * time.Second)
+			podname, err = util.Shell(`oc get pods -n openshift-operators -l name=istio-operator --field-selector spec.nodeName=%s -o jsonpath='{.items[0].metadata.name}'`, workername)
+			if err != nil {
+				t.Fatalf("Failed to get the operator pod name")
+			}
+		}
+		node, err := util.Shell(`oc get pods -n openshift-operators %s -o jsonpath='{.spec.nodeName}'`, podname)
+		if err != nil {
+			t.Fatalf("Failed to get the node name")
+		}
+		if node != workername {
+			t.Fatalf("Failed to run the operator on the infra node")
+		}
+		_, err = util.Shell(`oc label node %s node-role.kubernetes.io/infra-`, workername)
+		if err != nil {
+			t.Fatalf("Failed to unlabel the worker node as infra")
```

Comment:
@fjglira It's better to put those two unlabel commands in the cleanup function. 
Because the deferred cleanup always run when this test failed in the middle.

If you put those two here, execution may not reach this point if something unexpected or panic happened between the label node part and the unlabel node part above.